api/client/container/commit.go

```diff
@@ -0,0 +1,90 @@
+package container
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	"github.com/docker/engine-api/types"
+	containertypes "github.com/docker/engine-api/types/container"
+	"github.com/spf13/cobra"
+)
+
+type commitOptions struct {
+	container string
+	reference string
+
+	pause   bool
+	comment string
+	author  string
+	changes []string
+	config  string
+}
+
+// NewCommitCommand creats a new cobra.Command for `docker commit`
+func NewCommitCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts commitOptions
+
+	cmd := &cobra.Command{
+		Use:   "commit [OPTIONS] CONTAINER [REPOSITORY[:TAG]]",
+		Short: "Create a new image from a container's changes",
+		Args:  cli.RequiresMinMaxArgs(1, 2),
```

Comment:
Thanks @dnephin. `RequiresMinMaxArgs()` was added as part of the #23286. Now I have changed the name to `RequiresRangeArgs()` and updated the pull request.